Failing TestContractInitAndCompile due to unexpected go version

Go 1.18, Ubuntu, firstly discovered in https://github.com/nspcc-dev/neo-go/pull/3110 tests (https://github.com/nspcc-dev/neo-go/actions/runs/6467920919/job/17558942258?pr=3110).
```
2023-10-10T10:20:09.4189302Z === RUN   TestContractInitAndCompile/excessive_parameters#01
2023-10-10T10:20:09.4189485Z additional arguments given while this command expects none
2023-10-10T10:20:09.4233893Z ##[error]error while trying to compile smart contract file: /tmp/TestContractInitAndCompile2057603306/001/testcontract/main.go:13:27: undeclared name: any (requires version go1.18 or later)
2023-10-10T10:20:09.4241749Z === CONT  TestContractInitAndCompile
2023-10-10T10:20:09.4241926Z     executor.go:277: 
2023-10-10T10:20:09.4242586Z         	Error Trace:	/home/runner/work/neo-go/neo-go/internal/testcli/executor.go:277
2023-10-10T10:20:09.4243525Z         	            				/home/runner/work/neo-go/neo-go/cli/smartcontract/contract_test.go:281
2023-10-10T10:20:09.4243784Z         	Error:      	Received unexpected error:
2023-10-10T10:20:09.4244646Z ##[error]        	            	error while trying to compile smart contract file: /tmp/TestContractInitAndCompile2057603306/001/testcontract/main.go:13:27: undeclared name: any (requires version go1.18 or later)
2023-10-10T10:20:09.4245274Z         	Test:       	TestContractInitAndCompile
2023-10-10T10:20:09.4245525Z --- FAIL: TestContractInitAndCompile (0.06s)
2023-10-10T10:20:09.4245894Z     --- PASS: TestContractInitAndCompile/no_path_is_provided (0.00s)
2023-10-10T10:20:09.4246231Z     --- PASS: TestContractInitAndCompile/invalid_path (0.00s)
2023-10-10T10:20:09.4246605Z     --- PASS: TestContractInitAndCompile/excessive_parameters (0.00s)
2023-10-10T10:20:09.4247018Z     --- PASS: TestContractInitAndCompile/don't_rewrite_existing_directory (0.00s)
2023-10-10T10:20:09.4247358Z     --- PASS: TestContractInitAndCompile/missing_source (0.00s)
2023-10-10T10:20:09.4247774Z     --- PASS: TestContractInitAndCompile/missing_config,_but_require_manifest (0.00s)
2023-10-10T10:20:09.4248152Z     --- PASS: TestContractInitAndCompile/provided_non-existent_config (0.00s)
2023-10-10T10:20:09.4248540Z     --- PASS: TestContractInitAndCompile/provided_corrupted_config (0.00s)
2023-10-10T10:20:09.4248907Z     --- PASS: TestContractInitAndCompile/excessive_parameters#01 (0.00s)
```

--- a/cli/smartcontract/smart_contract.go
+++ b/cli/smartcontract/smart_contract.go
@@ -374,6 +374,9 @@ func initSmartContract(ctx *cli.Context) error {
 	}
 
 	gm := []byte("module " + contractName + `
+
+go 1.19
+
 require (
 	github.com/nspcc-dev/neo-go/pkg/interop ` + ver + `
 )`)

--- a/go.mod
+++ b/go.mod
@@ -1,5 +1,7 @@
 module github.com/nspcc-dev/neo-go
 
+go 1.19
+
 require (
 	github.com/chzyer/readline v1.5.1
 	github.com/consensys/gnark v0.9.1
@@ -76,5 +78,3 @@ require (
 	google.golang.org/protobuf v1.31.0 // indirect
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
-
-go 1.19

